packages/flutter_tools/lib/src/macos/cocoapods.dart

```diff
@@ -389,9 +389,129 @@ class CocoaPods {
+              "To build, increase your application's deployment target as "
+              'described at $docsLink\n\n'
+              'The minimum required version for "$sourcePlugin" could not be '
+              'determined. Please file an issue at '
+              'https://github.com/flutter/flutter/issues about this error message.',
+              emphasis: true,
+            );
+          }
+        } else {
+          // In theory this could find the failing pod's spec and parse out its
+          // minimum deployment version, but finding that spec would add a lot
+          // of complexity to handle a case that plugin authors should not
+          // create, so this just provides the actionable step of following up
+          // with the plugin developer.
+          _logger.printError(
+            'Error: The pod "${podInfo.failingPod}" required by the plugin '
+            '"$sourcePlugin" requires a higher minimum iOS deployment version '
+            "than the plugin's reported minimum version.\n"
+            'To build, remove the plugin "$sourcePlugin", or contact the plugin\'s '
+            'developers for assistance.',
+            emphasis: true,
+          );
+        }
+      }
     }
   }
 
+  ({String failingPod, String sourcePlugin, String podPluginSubdir})?
+      _parseMinDeploymentFailureInfo(String podInstallOutput) {
+    final RegExp sourceLine = RegExp(r'\(from `.*\.symlinks/plugins/([^/]+)/([^/]+)`\)');
+    final RegExp dependencyLine = RegExp(r'Specs satisfying the `([^ ]+).*` dependency were found, '
+        'but they required a higher minimum deployment target');
+    final RegExpMatch? sourceMatch = sourceLine.firstMatch(podInstallOutput);
+    final RegExpMatch? dependencyMatch = dependencyLine.firstMatch(podInstallOutput);
+    if (sourceMatch == null || dependencyMatch == null) {
+      return null;
+    }
+    return (
+      failingPod: dependencyMatch.group(1)!,
+      sourcePlugin:  sourceMatch.group(1)!,
+      podPluginSubdir: sourceMatch.group(2)!
+    );
+  }
+
+  String? _findPodspecMinDeploymentVersion(File podspec, String platformString) {
+    // There are two ways the deployment target can be specified; see
+    // https://guides.cocoapods.org/syntax/podspec.html#group_platform
+    final RegExp platformPattern = RegExp(
+      // Example: spec.platform = :osx, '10.8'
+      // where "spec" is an arbitrary variable name.
+      r'^\s*[a-zA-Z_]+\.platform\s*=\s*'
+      ':$platformString'
+      r'''\s*,\s*["']([^"']+)["']''',
+      multiLine: true
+    );
+    final RegExp deploymentTargetPlatform = RegExp(
+      // Example: spec.osx.deployment_target = '10.8'
+      // where "spec" is an arbitrary variable name.
+      r'^\s*[a-zA-Z_]+\.'
+      '$platformString\\.deployment_target'
+      r'''\s*=\s*["']([^"']+)["']''',
+      multiLine: true
+    );
+    final String podspecContents = podspec.readAsStringSync();
```

Comment:
You should validate the `podspec` file exists before you get its contents to avoid FileSystemException.

Reply:
Good point, added a check at the start of the method.